library/spec/Ivoz/Provider/Domain/Model/Invoice/InvoiceSpec.php

```diff
@@ -48,4 +50,94 @@ function it_is_initializable()
     {
         $this->shouldHaveType(Invoice::class);
     }
+
+
+    function it_resets_calculated_values_if_base_values_have_changed()
+    {
+        $dto = clone $this->dto;
+        $this->initChangelog();
+
+        $dto->setTotal(1);
+        $dto->setTaxRate(1);
+        $this->updateFromDto(
+            $dto,
+            new \spec\DtoToEntityFakeTransformer()
+        );
+
+        $this
+            ->getTotal()
+            ->shouldBe(null);
+    }
+
+    /**
+     * Value must be null so that is not automatically regenerated
+     */
+    function it_sets_null_status_on_reset()
+    {
+        $this->dto->setStatus(InvoiceInterface::STATUS_CREATED);
+        $dto = clone $this->dto;
```

Comment:
The other tests clone *before* setting status to `$this->dto`

Is the order in this test intended?

Reply:
Yes, $this->dto is used for entity instatiation and local $dto for entity update. I wanted them both to have created status :)